event-based-simulators/extras/GenerateMeasurements.py

```diff
@@ -0,0 +1,248 @@
+                    tenant_id=Environment.C8Y_TENANT,  # the tenant ID of your Cumulocity tenant here
+                    username=Environment.C8Y_USER,  # your Cumulocity IoT username
+                    password=Environment.C8Y_PASSWORD)  # your Cumulocity IoT password
+C8Y_HEADERS, MEASUREMENTS_HEADERS = ArgumentsAndCredentialsHandler.SetupHeadersForAPIRequest(
+    tenant_id=c8y.tenant_id, username=c8y.username, password=Environment.C8Y_PASSWORD)
+
+
+def current_timestamp(format="%Y-%m-%dT%H:%M:%S.%f"):
+    return datetime.utcnow().strftime(format)[:-3] + 'Z'
+
+
+logging.basicConfig(format='%(asctime)s %(name)s:%(message)s', level=logging.DEBUG)
+log = logging.getLogger("Measurements-generator")
+log.info(f"started at {current_timestamp()}")
+#####################################################
+'''
+# Check if connection to tenant can be created
+tenantConnectionResponse = ArgumentsAndCredentialsHandler.CheckTenantConnection(baseUrl=Environment.C8Y_BASE,
+                                                                                C8Y_HEADERS=C8Y_HEADERS)
+if tenantConnectionResponse:
+    log.info(f"Connect to tenant {Environment.C8Y_TENANT} successfully")
+else:
+    if tenantConnectionResponse is None:
+        log.error(f"Wrong base url setup. Check again the URL: {Environment.C8Y_BASE}")
+    else:
+        log.error(tenantConnectionResponse.json())
+    log.error(f"Connect to tenant {Environment.C8Y_TENANT} failed")
+    sys.exit()
+'''
+######################################################
+# Start and end date are mandatory and need to be in ISO8601 format (e.g. 2020-10-26T10:00:00.000).
+sim_time = datetime.utcnow()
+sim_end_time = sim_time + timedelta(hours=1)
+
+LIST_OF_MEASUREMENTS = ['Availability', 'AvailabilityLossTime', 'PerformanceLossTime', 'ActualProductionTime',
+                        'QualityLossAmount', 'IdealQualityTime', 'IdealCycleAmount', 'AvailabilityLossAmount',
+                        'IdealMachineRuntime', 'IdealProductionAmount', 'OEE', 'ActualQualityAmount',
+                        'ActualProductionAmount', 'IdealAmount', 'PerformanceLossAmount', 'Quality', 'IdealCycleTime',
+                        'QualityLossTime', 'Performance', 'PotentialProductionTime']
+
+
+def ListAllChildDevices():
+    deviceInTenantCount = 0
+    deviceManagedObject = c8y.device_inventory.select(type=C8Y_OEE_SIMULATOR_DEVICES_GROUP)
+    listOfChildDevicesExternalIds = []
+    for device in deviceManagedObject:
+        deviceInTenantCount += 1
+        log.info(
+            f"Found device '{device.name}', id: #{device.id}, owned by {device.owner}, number of children: {len(device.child_devices)}, type: {device.type}")
+        for childDevice in device.child_devices:
+            deviceExternalId, deviceExternalIdType = CheckDeviceExternalIdById(childDevice.id, c8y.base_url)
+            if not deviceExternalId:
+                return
```

Comment:
I would think to use continue, since return would discard the whole function run

Reply:
Yes, since I copy it from export script, I forget to fix it because the function is used to be an individual one and I would like to make the function end